### PR TITLE
Use new blobstore modules

### LINKED ISSUE #294: Add extensible support for cloud storage
- Terra will eventually support Azure.
- A general DRS client should support cloud copies to major cloud providers.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TLP-2083)

### PR DESCRIPTION
**Pull request by xbrianh**
closes #294

### CODE REVIEW

File: terra_notebook_utils/drs.py [hunk 1]
```diff
@@ -168,40 +175,36 @@ def resolve_drs_for_gs_storage(drs_url: str) -> Tuple[gs.Client, DRSInfo]:
...
     assert drs_url.startswith("drs://"), f'Not a DRS schema: {drs_url}'
     enable_requester_pays(workspace_name, workspace_namespace)
     try:
-        client, info = resolve_drs_for_gs_storage(drs_url)
-        blob = client.bucket(info.bucket_name, user_project=workspace_namespace).blob(info.key)
-        with gscio.Reader(blob, chunk_size=buffer) as handle:
-            the_bytes = handle.read(num_bytes)
-
+        blob = get_drs_blob(drs_url, workspace_namespace)
+        chunk_size = buffer or 2 * num_bytes
```
Comment (DailyDreaming) [inline terra_notebook_utils/drs.py:205]: Is the default buffer size always twice as big as the actual data it's fetching (`2 * num_bytes`)?  Would this still work if it just defaults to the size of the data (`num_bytes`)?

File: terra_notebook_utils/cli/drs.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def drs_copy_batch(args: argparse.Namespace):
 @drs_cli.command("head", arguments={
     "drs_url": dict(type=str),
     "--bytes": dict(type=int, required=False, default=1, help="Number of bytes to fetch."),
-    "--buffer": dict(type=int, required=False, default=MULTIPART_THRESHOLD,
+    "--buffer": dict(type=int, required=False, default=None,
```
Comment (DailyDreaming) [inline terra_notebook_utils/cli/drs.py:63]: Could the default value (or the description that all data will be loaded without a buffer) be included in the `help`?

**Review by DailyDreaming — APPROVED** [review]
LGTM.  See comments.

File: terra_notebook_utils/cli/drs.py (hunk 2 above)
Comment (xbrianh) [reply terra_notebook_utils/cli/drs.py:63]: The size of `buffer` is no longer meaningful.

This should be revisited in a future PR. I will file an issue.

**Pull request merged**